process-data/hxl_parser.py

```diff
@@ -21,21 +21,86 @@ def process_hxl_files(file_path):
     for sheet_name in xlsx_file.sheet_names:
         sheet_data = pd.read_excel(file_path, sheet_name=sheet_name, header=None)
         clean_sheet = clean_one_sheet(sheet_data)
-        if clean_sheet:
+        if not clean_sheet is None:
             processed_data_list.append(clean_sheet)
 
     return processed_data_list
 
+#check if the dataset contains aleast one row with numeric values
+def numeric_row(df, row_index, cols):
+    if row_index + 1 >= df.size:
+        return None
+
+    cols_numeric = list(filter(lambda x : numeric_regex.match(str(df[x].loc[row_index + 1])), cols.keys())) 
+    if not cols_numeric:
+        return None
+    
+    return cols_numeric
+
+#remove aggregation on a specific state level
+def clean_adm_cols(df):
+    adm_cols = list(filter(lambda x : x.startswith("#adm"), df.columns.astype(str)))
+    if not adm_cols:
+        return df
+    for col in adm_cols[: -1]:
+        df[col] = df[col].ffill()
+    return df.dropna(subset = adm_cols)
+
+def contains_affected(numeric_cols, hxl_indices):
+    # derived from https://stackoverflow.com/a/2364277
+    return next((i for i, col_index in enumerate(numeric_cols) if hxl_indices[col_index].startswith("#affected")), None)
+
+# add preceeding rows if tags are in the middle of the data frame
+def add_preceeding_rows(df, aff_ind, i, numeric_columns):
+    ii = i - 1
```

Comment:
could you please rename the variable, insteal of i and ii do more meaningful names (for readability and maintainability)